### ISSUE TITLE
[Bug]: aws_lambda_function env vars update not detected when dependent on new resources' output

### ISSUE THREAD

**Issue by sehyod** [hashicorp/terraform-provider-aws#39572]
### Terraform Core Version

1.9.7

### AWS Provider Version

5.69.0

### Affected Resource(s)

* aws_lambda_function

### Expected Behavior

Adding env vars depending on the output of new resources should be detected in the plan phase.

Something like 
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create
  ~ update in-place

Terraform will perform the following actions:

  # aws_lambda_function.test_lambda will be updated in-place
  ~ resource "aws_lambda_function" "test_lambda" {
        id                             = "test"
        tags                           = {}
        # (29 unchanged attributes hidden)

      + environment {
          + variables = {
              + "PLAIN_TEXT_ENVVAR" = "value"
              + "DYNAMIC_ENVVAR"    = (known after apply)
            }
        }

        # (3 unchanged blocks hidden)
    }

  # null_resource.test_resource will be created
  + resource "null_resource" "test_resource" {
      + id = (known after apply)
    }

Plan: 1 to add, 1 to change, 0 to destroy.
```

### Actual Behavior

The env vars update is not detected:
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # null_resource.test_resource will be created
  + resource "null_resource" "test_resource" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

### Relevant Error/Panic Output Snippet

```shell
Error: Provider produced inconsistent final plan
│
│ When expanding the plan for aws_lambda_function.test_lambda to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid new value for .environment: block
│ count changed from 0 to 1.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

### Terraform Configuration Files

```terraform
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "5.69.0"
    }
    null = {
      source  = "hashicorp/null"
      version = "3.2.3"
    }
  }
}

data "aws_iam_policy_document" "assume_role" {
  statement {
    actions = ["sts:AssumeRole"]
    principals {
      type        = "Service"
      identifiers = ["lambda.amazonaws.com"]
    }
  }
}

resource "aws_iam_role" "test_role" {
  name               = "test-role"
  assume_role_policy = data.aws_iam_policy_document.assume_role.json
}

# resource "null_resource" "test_resource" {}

resource "aws_lambda_function" "test_lambda" {
  function_name = "test"

  role = aws_iam_role.test_role.arn

  filename = "code.zip"
  handler  = "main.handler"

  runtime = "python3.11"

  # environment {
  #   variables = {
  #     PLAIN_TEXT_ENVVAR = "value"
  #     DYNAMIC_ENVVAR    = null_resource.test_resource.id
  #   }
  # }

  memory_size = 1024
  timeout     = 60
}
```

### Steps to Reproduce

1. Run `terraform init`
2. Run `terraform apply`
3. Uncomment the `null_resource` and the `environment` block
4. Run `terraform plan`
     - Only the new `null_resource` is being detected
5. Change something in the lambda config: set the `memory_size` to 512 for example
6. Run `terraform plan`
     - The memory change is detected but not the environment block
7. Run `terraform apply`
     - This results in the `provider produced an invalid new value for .environment: block` shared above
8. Run `terraform plan` again
     - This time the environment block change is being properly detected because the `null_resource` has been created during the last run

NB: If you run `terraform apply` after step 4, you won't get an error, but the lambda function will be created without its environment variables. You then need to rerun the command to actually add the env vars

### Would you like to implement a fix?

No

**Comment by justinretzolk** [comment]
Hey @sehyod 👋 Thank you for taking the time to raise this! It looks like this was already noticed in #38755. We like to try to keep conversations consolidated where possible, so I'm going to close this issue in favor of the existing one.

**Comment by justinretzolk** [comment]
Duplicate of #38755
